lib/maths/CTimeSeriesDecompositionDetail.cc

```diff
@@ -1561,15 +1611,14 @@ bool CTimeSeriesDecompositionDetail::CComponents::addSeasonalComponents(
                     time += dt;
                 }
             }
-            if (result.piecewiseLinearTrend()) {
-                LOG_TRACE(<< "Piecewise linear trend");
-                values = CTimeSeriesSegmentation::removePiecewiseLinear(
-                    values, CTimeSeriesSegmentation::piecewiseLinear(values));
-            }
             if (piecewiseConstantLinearScaling) {
-                LOG_TRACE(<< "Piecewise constant linear scaling");
                 this->adjustValuesForPiecewiseConstantScaling(period / dt, values);
             }
+            for (std::size_t i = 0; i < values.size(); ++i) {
+                if (CBasicStatistics::count(values[i]) > 0.0) {
+                    CBasicStatistics::moment<0>(values[i]) += shifts[i];
```

Comment:
It returns by constant reference. This is mainly to be consistent with higher moments which must return by value since they manipulate the value stored. `moment` is intended to be used for writable access.